### PR TITLE
Upgrade bs-platform and fix Js.bool related fixes

### PR DESCRIPTION
**Pull request by Gregoirevda**
I did a `yarn add -D bs-platform` which is now on `3.0.0` and it has support for Js.booleans natively.
I now have a lot of warnings of unnecessary Js.bool casting functions

### CODE REVIEW

File: src/private/encode.re [hunk 1]
```diff
@@ -8,7 +8,7 @@ external float : float => Js.Json.t = "%identity"; /* [float n] makes a JSON num
 external int : int => Js.Json.t = "%identity"; /* [int n] makes a JSON number of the [int] [n] */
 
 /** [boolean b] makes a JSON boolean of the [Js.boolean] [b] */
-external boolean : Js.boolean => Js.Json.t = "%identity"; /* [boolean b] makes a JSON boolean of the [Js.boolean] [b] */
+external boolean : bool => Js.Json.t = "%identity"; /* [boolean b] makes a JSON boolean of the [Js.boolean] [b] */
```
Comment (Gregoirevda) [inline src/private/encode.re]: Not sure if the comment is still ok

File: src/components/textInput.re [hunk 2]
```diff
@@ -1,9 +1,8 @@
 [@bs.module "react-native"]
 external view : ReasonReact.reactClass = "TextInput";
 
-[@bs.send] external _isFocused : ReasonReact.reactRef => Js.boolean = "";
+[@bs.send] external isFocused : ReasonReact.reactRef => bool = "";
 
-let isFocused = UtilsRN.(Js.to_bool << _isFocused);
```
Comment (Gregoirevda) [inline src/components/textInput.re]: Please double check this one, not sure what UtilsRN. is doing

Comment (grsabreu) [reply src/components/textInput.re]: I think it is just doing pointfree composition

File: src/private/encode.re (hunk 1 above)
Comment (wokalski) [reply src/private/encode.re]: It should be [boolean b] makes a JSON boolean of the [bool] [b]

**Comment by grsabreu** [comment]
Should we bump the bindings to 1.0.0? As this is a breaking change for previous bs-platform versions
cc @wokalski

**Comment by wokalski** [comment]
yes, a bindings release is long overdue. I think we should 😀. I'm sort of bummed out because we planned to have migration scripts for all breaking changes but we still don't have infrastructure for that @grsabreu .

**Comment by gunnigylfa** [comment]
Would it be possible to use the reason-react migration script? https://github.com/chenglou/upgrade-reason-react
If that works then we could just add that step to the release notes.

**Comment by vnovick** [comment]
It's nice to have a migration script, but does not having it really blocks us from migrating the change?
Js.boolean change to bool according to @chenglou and @jaredly are not a breaking change. 
@chenglou , @jaredly , What's your insights on this?

**Comment by arnarthor** [comment]
I agree. It would be great to have a migration script, but I don't think it should be a blocker for updating this.

**Comment by chenglou** [comment]
The migration script would be going the extra mile, which is great! This one’s simple though. Could be a few regexes. The heavier migration infra is also in place, and we’ve used it twice for ReasonReact already, so no worries. But again, this can just be a regex change. 

The Boolean changes isn’t breaking (all previous helpers are now no-ops).

**Comment by wokalski** [comment]
@chenglou thankfully we don't need your permission ;).

In any case, let's merge it.

**Comment by chenglou** [comment]
I’m just chiming in since this seems to have stalled a bit

**Comment by wokalski** [comment]
My comment was unnecessary.

I got biased by your controlling behaviour in person and this was an impolite intuitional answer. Anyway, I need to set the standards higher.

**Comment by chenglou** [comment]
All good. I commented because another member from the community asked me to

**Pull request merged**